haystack/components/rankers/transformers_similarity.py

```diff
@@ -74,8 +80,19 @@ def warm_up(self):
         Warm up the model and tokenizer used for scoring the Documents.
         """
         if self.model_name_or_path and not self.model:
-            self.model = AutoModelForSequenceClassification.from_pretrained(self.model_name_or_path, token=self.token)
-            self.model = self.model.to(self.device)
+            if torch.cuda.is_available():
```

Comment:
This if/elif/else code block from here to line 92 is the same as in the ExtractiveReader and it could be reused probably also in other future components. I suggest that we create a method for this block in haystack/utils, basically containing
```python
            if torch.cuda.is_available():
                self.device = self.device or "cuda:0"
            elif (
                hasattr(torch.backends, "mps")
                and torch.backends.mps.is_available()
                and os.getenv("HAYSTACK_MPS_ENABLED", "true") != "false"
            ):
                self.device = self.device or "mps:0"
            else:
                self.device = self.device or "cpu:0"
```

Reply:
I made a util function for this called `get_device` and updated the ExtractiveReader to use it as well. I was also able to add some unit tests checking most of the logic in that function.